builder/vsphere/supervisor/step_watch_source.go

```diff
@@ -112,6 +116,8 @@ func (s *StepWatchSource) initStep(state multistep.StateBag) error {
 	s.SourceName = sourceName
 	s.Namespace = namespace
 	s.KubeWatchClient = kubeWatchClient
+	s.timeoutCh = time.After(time.Duration(s.Config.WatchSourceTimeoutSec) * time.Second)
```

Comment:
I believe that storing this during the initStep actually starts the timer, which mean that reusing it won't actually give you the full duration. Have you tested this with a smaller duration to see how it behaves?